File: 136-single-number/single-number.go
func singleNumber(nums []int) int {
    if len(nums) == 1 {
        return nums[0]
    }
    sort.Ints(nums)
    for i := 0; i < len(nums); i++ {
        if i + 1 < len(nums) && nums[i] == nums[i + 1] {
            i++
        } else {
            return nums[i]
        }
    }
    return 0
}

File: 136-single-number/single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        if (nums.size() == 1) {
            return nums[0];
        }
        sort(nums.begin(), nums.end());
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i + 1] == nums[i]) {
                i++;
            } else {
                return nums[i];
            }
        }
    return 0;
    }
};